service/git/service.go

```diff
@@ -50,6 +51,49 @@ func validateReferences(ctx context.Context, validateRefName bool, refs ...*look
 	return nil
 }
 
+func (r *Service) getFrom(
+	ctx context.Context,
+	base, head *lookout.ReferencePointer,
+) (
+	*lookout.ReferencePointer, error,
+) {
+	commits, storer, err := r.loader.LoadCommits(ctx, *base, *head)
+	if err != nil {
+		return nil, err
+	}
+
+	var commonAncestor *object.Commit
+	res, err := merge_base.MergeBase(storer, commits[0], commits[1])
+	if err != nil {
+		return nil, err
+	}
+
+	if len(res) == 0 {
+		// If there is no common ancestor between two commits it means that they
+		// don't have a common history. That PR won't be able to be created in
+		// GitHub, but in other situations (ie. with lookout-sdk), an analyzer
+		// could be interested in analyzing the difference between both commints.
+		return base, nil
```

Comment:
why not return an error? If you ask for the common ancestor and it does not exist, maybe returning `base` makes other things fail later, without any hint of why.

Reply:
An analyzer can be interested on reviewing changes between orphan branches. Couldn't it?

Reply:
what's the behaviour of `git diff base...head` if they do not have a common ancestor?

Reply:
I tried locally, and `git diff base...head` behaves as `..` if there is no shared history between them?

For curiosity, I checked `git/git` [source code, and it seems](https://github.com/git/git/blob/master/builtin/diff.c#L459) that `...` behaves that way only if there is at least one "merge-base" between `base` and `head`; otherwise it seems to fallback as `..`. Anyhow I'm not used to `git` codebase.